plugins/modules/pnp_intent.py

```diff
@@ -763,49 +879,47 @@ def get_have(self):
                 function='get_software_image_details',
                 params=self.want.get("image_params"),
             )
-
-            self.log(str(image_response))
-
             image_list = image_response.get("response")
+            self.log(str(image_response))
 
-            if len(image_list) == 1:
-                have["image_id"] = image_list[0].get("imageUuid")
-                self.log("Image Id: " + str(have["image_id"]))
-            else:
-                self.module.fail_json(msg="Image not found", response=[])
-
-            # check if given template exists, if exists store template id
+            # check if project has templates or not
             template_list = self.dnac_apply['exec'](
                 family="configuration_templates",
                 function='gets_the_templates_available',
                 params={"project_names": self.want.get("project_name")},
             )
-
             self.log(str(template_list))
 
-            if template_list and isinstance(template_list, list):
-                # API execution error returns a dict
-                template_details = get_dict_result(template_list,
-                                                   'name', self.want.get("template_name"))
-                if template_details:
-                    have["template_id"] = template_details.get("templateId")
-
-                    self.log("Template Id: " + str(have["template_id"]))
-                else:
-                    self.module.fail_json(msg="Template not found", response=[])
-            else:
-                self.module.fail_json(msg="Project Not Found", response=[])
-
             # check if given site exits, if exists store current site info
-            site_name = self.want.get("site_name")
-
             site_exists = False
-            (site_exists, site_id) = self.site_exists()
+            if isinstance(self.want.get("site_name"), str):
```

Comment:
if not isinstance, then you can fail and return. then you can follow the flow.
Or
If you want to check isinstance of str is true, then you can move the "if site_exists" inside the isinstance..